exes/mqtt-bridge/inc/run.hpp

```diff
@@ -27,64 +30,58 @@ class run {
 
   auto start() -> asio::awaitable<void> {
     while (true) {
+      restart_needed_ = false;
+
       logger.trace("----------------------------------------------------------------------------");
       logger.trace("Event loop started");
 
-      bool connection_success = co_await sp_interface_.connect_mqtt_client();
+      sp_interface_.emplace(io_ctx_, config_);
 
-      if (!connection_success) {
+      if (!co_await sp_interface_->connect_mqtt_client()) {
         continue;
       }
 
-      bool subscribe_success = co_await sp_interface_.subscribe_to_ncmd();
-
-      if (!subscribe_success) {
+      if (!co_await sp_interface_->subscribe_to_ncmd()) {
         continue;
       }
 
-      exter_to_tfc_.create_outward_signals();
-
-      tfc_to_exter_.set_signals();
+      exter_to_tfc_.emplace(io_ctx_, config_, ipc_client_);
+      exter_to_tfc_->create_outward_signals();
 
-      sp_interface_.set_value_change_callback(std::bind_front(&ext_to_tfc::receive_new_value, &exter_to_tfc_));
+      tfc_to_exter_.emplace(io_ctx_, sp_interface_.value(), ipc_client_, config_, restart_needed_);
+      tfc_to_exter_->set_signals();
 
-      co_await asio::steady_timer{ io_ctx_, std::chrono::seconds{ 1 } }.async_wait(asio::use_awaitable);
+      sp_interface_->set_value_change_callback(std::bind_front(&ext_to_tfc::receive_new_value, &exter_to_tfc_.value()));
 
       asio::cancellation_signal cancel_signal{};
 
-      bool restart_needed = false;
+      co_spawn(sp_interface_->strand(),
+               sp_interface_->wait_for_payloads(std::bind_front(&spark_plug::process_payload, &sp_interface_.value()),
+                                                restart_needed_),
+               bind_cancellation_slot(cancel_signal.slot(), asio::detached));
 
-      co_spawn(
-          sp_interface_.strand(),
-          sp_interface_.wait_for_payloads(std::bind_front(&spark_plug::process_payload, &sp_interface_), restart_needed),
-          bind_cancellation_slot(cancel_signal.slot(), asio::detached));
-
-      io_ctx_.run_for(std::chrono::seconds{ 1 });
-
-      while (!restart_needed) {
-        co_await asio::steady_timer{ sp_interface_.strand(), std::chrono::seconds{ 5 } }.async_wait(asio::use_awaitable);
+      while (!restart_needed_) {
+        co_await asio::steady_timer{ sp_interface_->strand(), std::chrono::milliseconds{ 100 } }.async_wait(
+            asio::use_awaitable);
       }
```

Comment:
Looking into this and will ping this thread when I have a solution, are you talking about `boost::asio::completion_token`?

Reply:
This can be implemented for reloading the application based on new signals, there is another reloading which happens when an incorrect packet is received. In my experience, when that happens the program goes into an endless loop and needs to be restarted anyways so I removed that. If that becomes a problem in the future it needs to be investigated on its own.